src/Compilers/CSharp/Portable/BoundTree/BoundNodeExtensions.cs

```diff
@@ -94,5 +97,163 @@ private class ContainsAwaitVisitor : BoundTreeWalkerWithStackGuardWithoutRecursi
                 return null;
             }
         }
+
+        public static bool VisitBinaryOperatorInterpolatedString<TArg, TInterpolatedStringType>(
```

Comment:
nit: It may be good to document order of visit. For the leaves, we call `visitor` from left to right. For the binary nodes, we call `binaryOperatorCallback in depth-first postorder if I understood correctly?
I think a recursive implementation would have been easier to follow, but I assume you're worried about recursion depth.